utils/search/search_general.py

```diff
@@ -22,19 +22,356 @@
 
 logger = logging.getLogger("search")
 console_logger = logging.getLogger("console")
 
 
+def search_prepare_sort(sort, options):
+    """Creates sort list for ordering by rating.
+
+    Order by rating, then by number of ratings.
+    
+    Args:
+        sort (str): sort url query parameter.
+        options (List[Tuple(str)]): list containing 2-element tuples with the ordering option names
+        and their corresponding fields in the database. 
+
+    Returns:
+        List[str]: list containing the sorting parameters.
+    """
+    if sort in [x[1] for x in options]:
+        if sort == "avg_rating desc":
+            sort = [sort, "num_ratings desc"]
+        elif  sort == "avg_rating asc":
+            sort = [sort, "num_ratings asc"]
+        else:
+            sort = [sort]
+    else:
+        sort = [forms.SEARCH_DEFAULT_SORT]
+    return sort
+
+
+def search_process_filter(filter_query):
+    """Process the filter to replace humnan-readable Audio Commons descriptor names.
+
+    Used for the dynamic field names used in Solr (e.g. ac_tonality -> ac_tonality_s, ac_tempo -> ac_tempo_i).
+    The dynamic field names we define in Solr schema are '*_b' (for bool), '*_d' (for float), '*_i' (for integer) 
+    and '*_s' (for string). At indexing time we append these suffixes to the ac descirptor names so Solr can 
+    treat the types properly. Now we automatically append the suffices to the filter names so users do not 
+    need to deal with that.
+
+    Args:
+        filter_query (str): query filter string.
+
+    Returns:
+        str: processed filter query string.
+    """
+    for name, t in settings.AUDIOCOMMONS_INCLUDED_DESCRIPTOR_NAMES_TYPES:
+        filter_query = filter_query.replace('ac_{0}:'.format(name), 'ac_{0}{1}:'
+                                            .format(name, settings.SOLR_DYNAMIC_FIELDS_SUFFIX_MAP[t]))
+    return filter_query
+
+
+def split_filter_query(filter_query, cluster_id):
+    """Parses and pre-process search filter parameters and returns the filters' information.
+
+    This function is used in the search template to display the filter and the link when removing them.
+    The clustering facet filter is in a separate query parameter. It facilitates to distinguish them from
+    the classical facet filters.
+
+    Args:
+        filter_query (str): query filter string.
+        cluster_id (str): cluster filter string.
+
+    Returns:
+        List[dict]: list of dictionaries containing the filter name and the url when removing the filter.
+    """
+    # Generate array with information of filters
+    filter_query_split = []
+    if filter_query != "":
+        for filter_str in re.findall(r'[\w-]+:\"[^\"]+', filter_query):
+            valid_filter = True
+            filter_str = filter_str + '"'
+            filter_display = filter_str.replace('"', '')
+            filter_name = filter_str.split(":")[0]
+            if filter_name != "duration" and filter_name != "is_geotagged":
```

Comment:
This parsing code is so frightening... It deserves a refactor I think but that's be for a different PR.